apiserver/src/test/java/org/dependencytrack/persistence/jdbi/ComponentDaoTest.java

```diff
@@ -159,6 +155,9 @@ public void testCascadeDeleteComponent() {
         assertThatNoException().isThrownBy(() -> qm.getObjectById(Vulnerability.class, vuln.getId()));
         assertThatNoException().isThrownBy(() -> qm.getObjectById(PolicyCondition.class, policyCondition.getId()));
         assertThatNoException().isThrownBy(() -> qm.getObjectById(Policy.class, policy.getId()));
+
+        // Ensure that metrics have been deleted.
+        assertThat(withJdbiHandle(handle ->  handle.attach(MetricsDao.class).getDependencyMetricsSince(component.getId(), Instant.now())).isEmpty());
```

Comment:
`Instant.now()` in this line may end up being a later point in time than `Instant.now()` when the `DependencyMetrics` record gets created. 

The `Instant` used here should be in the past to ensure it would technically include the metrics created earlier.